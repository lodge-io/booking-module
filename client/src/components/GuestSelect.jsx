const { React } = window;

const GuestSelectDiv = styled.div`
  border-radius: 2px;
  padding:10px;
`;

const GuestTypeRow = styled.div`
  display: flex;
  padding:10px;
`;

const GuestRowElement = styled.div`
  width:33%;
`;

const RoundButton = styled.button`
  border-radius:15px;
  background-color:white;
  color:#00777B;
  border: 1px solid #00777B;
  width:30px;
  height:30px;
`;

const CloseElement = styled.div`
  text-align:right;
  color: #00777B;
  &:hover{
    text-decoration: underline;
  }
`;


const GuestTypeEntry = ({ type, count, setGuestCount }) => (
  <GuestTypeRow>
    <GuestRowElement>{type}</GuestRowElement>
    <GuestRowElement />
    <GuestRowElement>
      <RoundButton type="button" onClick={() => setGuestCount(type.toLowerCase(), count - 1)}>-</RoundButton>
      {`  ${count}  `}
      <RoundButton type="button" onClick={() => setGuestCount(type.toLowerCase(), count + 1)}>+</RoundButton>
    </GuestRowElement>
  </GuestTypeRow>
);

const GuestSelect = ({ guests, setGuestCount, close }) => {
  return (
    <GuestSelectDiv>
      <GuestTypeEntry type="Adults" count={guests.adults} setGuestCount={setGuestCount} />
      <GuestTypeEntry type="Children" count={guests.children} setGuestCount={setGuestCount} />
      <GuestTypeEntry type="Infants" count={guests.infants} setGuestCount={setGuestCount} />
      <CloseElement onClick={close}>close</CloseElement>
    </GuestSelectDiv>
  );
};



export default GuestSelect;
