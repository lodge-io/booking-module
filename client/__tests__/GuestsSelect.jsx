import React from 'react';
import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';
import GuestSelect from '../src/components/GuestSelect';

function makeGuestObj() {
  return { adults: 1, children: 0, infants: 3 };
}

describe('GuestSelect', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<GuestSelect guests={makeGuestObj()} setGuestCount={() => {}} close={() => {}} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  xit('should invoke setGuestCount appropriately when button clicked', () => {

  });
});
