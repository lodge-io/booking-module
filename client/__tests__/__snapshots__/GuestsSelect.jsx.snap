// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GuestSelect renders correctly 1`] = `
<div
  className="sc-bdVaJa hyCpos"
>
  <div
    className="sc-bwzfXH jIEWCi"
  >
    <div>
      Adults
    </div>
    <div />
    <div>
      <button
        onClick={[Function]}
        type="button"
      >
        -
      </button>
      1
      <button
        onClick={[Function]}
        type="button"
      >
        +
      </button>
    </div>
  </div>
  <div
    className="sc-bwzfXH jIEWCi"
  >
    <div>
      Children
    </div>
    <div />
    <div>
      <button
        onClick={[Function]}
        type="button"
      >
        -
      </button>
      0
      <button
        onClick={[Function]}
        type="button"
      >
        +
      </button>
    </div>
  </div>
  <div
    className="sc-bwzfXH jIEWCi"
  >
    <div>
      Infants
    </div>
    <div />
    <div>
      <button
        onClick={[Function]}
        type="button"
      >
        -
      </button>
      3
      <button
        onClick={[Function]}
        type="button"
      >
        +
      </button>
    </div>
  </div>
  <div
    onClick={[Function]}
  >
    close
  </div>
</div>
`;
